src/core/ddsc/src/dds_topic.c

```diff
@@ -583,6 +583,30 @@ dds_entity_t dds_create_topic_impl (
     struct dds_psmx *psmx = dom->psmx_instances.instances[i];
     if (!psmx->ops.type_qos_supported (psmx, DDS_PSMX_ENDPOINT_TYPE_UNSET, sertype_registered->data_type_props, new_qos))
       continue;
+
+    // Check if the topic is in the forbiddenTopics or NOT in the onlyForTopics list, depending
+    // on which one is used.
+    // First off, topics not mentioned are allowed iff only_for_topics is unused
+    bool allowed_by_config = !psmx->only_for_topics[0];
+    // Then any topic in only_for_topics is allowed
+    for (char **topic = psmx->only_for_topics; *topic; topic++) {
+      if (strcmp(ktp->name, *topic) == 0) {
```

Comment:
I would also use pattern matching (perhaps even regex matching — but those are not as portable), not just exact matches. It is only used outside the "hot" path, so performance is not that big of an issue.

There is https://github.com/eclipse-cyclonedds/cyclonedds/blob/00c63c342c2d858bf94b6a602dcf8bd696fdbe26/src/core/ddsi/src/ddsi_misc.c#L39, and there is also https://github.com/eclipse-cyclonedds/cyclonedds/blob/00c63c342c2d858bf94b6a602dcf8bd696fdbe26/src/core/ddsi/src/ddsi_nwpart.c#L92 which at the very least has a passing resemblance in functionality to the lists here.

Reply:
@eboasson Done. There was a slight complication in that `ddsi_misc` was not available from the place where I needed it, so I added a new header `ddsi/ddsi_misc.h` to export it.